ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Ledger/Tables/MapKind.hs

```diff
@@ -0,0 +1,216 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE QuantifiedConstraints      #-}
+{-# LANGUAGE RankNTypes                 #-}
+{-# LANGUAGE StandaloneKindSignatures   #-}
+{-# LANGUAGE TypeFamilies               #-}
+
+module Ouroboros.Consensus.Ledger.Tables.MapKind (
+    -- * Classes
+    CanEmptyMK (..)
+  , CanMapKeysMK (..)
+  , CanMapMK (..)
+  , EqMK
+  , NoThunksMK
+  , ShowMK
+    -- * Concrete MapKinds
+  , Canonical (..)
+  , CodecMK (..)
+  , DiffMK (..)
+  , EmptyMK (..)
+  , KeysMK (..)
+  , SeqDiffMK (..)
+  , TrackingMK (..)
+  , ValuesMK (..)
+  ) where
+
+import qualified Codec.CBOR.Decoding as CBOR
+import qualified Codec.CBOR.Encoding as CBOR
+import           Data.Kind (Constraint)
+import           Data.Map.Diff.Strict (Diff)
+import qualified Data.Map.Diff.Strict.Internal as Diff.Internal
+import           Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import           Data.Set (Set)
+import qualified Data.Set as Set
+import           GHC.Generics (Generic)
+import           NoThunks.Class
+import           Ouroboros.Consensus.Ledger.Tables.Common
+import           Ouroboros.Consensus.Ledger.Tables.DiffSeq
+
+{-------------------------------------------------------------------------------
+  Classes
+-------------------------------------------------------------------------------}
+
+type CanEmptyMK :: MapKind -> Constraint
+class CanEmptyMK mk where
```

Comment:
Perhaps use `NullableMK` instead of `CanEmpty`? Or `NullMK`

Reply:
We chose `ZeroableMK`